Update for Nougat compatibility

mfonville opened the pull request.

mfonville committed:
Update oat2dex.jar to 031a256

mfonville committed:
Fix the script and update oat2dex for Nougat (2.2b3)

Lekensteyn commented on odex2apk.py:
```diff
@@ -204,24 +204,25 @@ def detect_paths(apk_file, arch=None, framework_path=None):
     # Assumed boot path, could be non-existing and be created later.
-    boot_odex_path = os.path.join(framework_path, arch, "odex")
+    boot_path = os.path.join(framework_path, arch)
+    
```
remove excess newline?

Lekensteyn commented on odex2apk.py:
```diff
@@ -204,24 +204,25 @@ def detect_paths(apk_file, arch=None, framework_path=None):
             format="%(name)s: %(message)s")
 
-    arch, boot_odex_path = detect_paths(args.apk_files[0],
+    arch, boot_path= detect_paths(args.apk_files[0],
```
missing space before `=`

Lekensteyn reviewed (commented):
Do you have an example (file listing) where the current version failed? It would be great if you could add it to the commit message such that it is clear how the change works. Thanks!

mfonville commented:
the path-expectations of oat2dex have changed, because (bak)smali has changed its interface.
To find out the differences fixed in smali for N, this is a good place to start: https://github.com/JesusFreke/smali/issues/418

I will fix the style comments

Lekensteyn commented on odex2apk.py:
```diff
@@ -204,24 +204,24 @@ def detect_paths(apk_file, arch=None, framework_path=None):
             format="%(name)s: %(message)s")
 
-    arch, boot_odex_path = detect_paths(args.apk_files[0],
+    arch, boot_path  = detect_paths(args.apk_files[0],
```
Now there are two spaces here before `=`, have I overlooked something?

mfonville replied:
lol whut, I will check and fix :P
not sure how that happened

Lekensteyn commented:
When you say that the oat2dex path expectations have changed, do you mean that the included .jarfile now produces a different location? Do the comments on top of the file needs an update to reflect new paths?

mfonville commented:
I removed the double spaces, and updated the documentation at the top a little bit, to reflect the changes that were already introduced with Marshmallow.

One big change is, that technically not all boot-stuff has to be de-odexed anymore, before the APKs/frameworks can be. But I am not 100% sure (not a real expert in this field).

Lekensteyn commented:
I noticed that you update the oat2dex.jar twice, is there anything special with that?

Where did you get the jar from? I'll try reproduce it like I did in e55098ec07e35ccbd71daea165a972f53b9daecf

mfonville commented:
@Lekensteyn I had it updated in an intermediate version, it ain't important but just part of my commit history...

Lekensteyn commented:
Well, the file is 1.3MB, so if possible I would like to keep it out of the history to reduce the clone size ;-)

What commit is your jar file based on?

mfonville commented:
Current jar is https://github.com/testwhat/SmaliEx/releases/tag/snapshot-s

Lekensteyn commented:
Ok, would you mind if I wait a week before merging (in case the snapshot results in issues)?

mfonville commented:
That's fine ofc; there is no hurry and the PR is mostly to at least notify you as upstream of these changes

mfonville committed:
Update oat2dex to 2.2b4

mfonville committed:
Update to Smali 2.2.1 oat2dex

mfonville committed:
Update oat2dex to f2f856e of testwhat
https://github.com/testwhat/SmaliEx/commit/f2f856eda4d580f5a3bd79bfe5f17da473c66c56

mfonville committed:
Update oat2dex
https://github.com/mfonville/SmaliEx/commit/620c55051e03458a80e98e3f2e82d60cd0448435

mfonville committed:
Update oat2dex
to https://github.com/mfonville/SmaliEx/commit/6b672c19a41ce9eb5cb5065c744ab2d8fcfd4af3

mfonville committed:
Update oat2dex to https://github.com/testwhat/SmaliEx/commit/ea9d5c42b6e3bdc151a89c84b58e493f691d0ac4
